pkg/volume/plugins.go

```diff
@@ -121,6 +122,9 @@ type VolumeHost interface {
 	// the provided spec.  See comments on NewWrapperBuilder for more
 	// context.
 	NewWrapperCleaner(spec *Spec, podUID types.UID, mounter mount.Interface) (Cleaner, error)
+
+	//Get cloud provider from kubelet
+	GetCloudProvider() cloudprovider.Interface
```

Comment:
> The passing of configs should not be copied all over.

@thockin can you explain this further?  I proposed volumeConfig in #9870 which is extensible to any type of config a volume plugin would need.  I see no reason it couldn't pass an OpenStack config in it.  Each volume takes what it needs from the volumeConfig struct.  

Alternatively, volumeConfig as a struct can be nuked and all config options are passed via VolumeHost.  I originally did it that way but was told that passing config down into components was the preferred method.

I think we should adopt 1 way and use it for all volume configuration.